xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/transactions/SetFee.java

```diff
@@ -53,34 +57,107 @@ static ImmutableSetFee.Builder builder() {
    * The charge, in drops of XRP, for the reference transaction, as hex. (This is the transaction cost before scaling
    * for load.)
    *
-   * @return A hex {@link String} basefee value.
+   * @return A hex {@link String} baseFee value.
+   *
+   * @deprecated Prefer {@link #baseFeeDrops()} over this field.
    */
-  @JsonProperty("BaseFee")
-  String baseFee();
+  @Value.Default
+  @Deprecated
+  @JsonIgnore
+  default String baseFee() {
+    return baseFeeDrops().value().toString(16);
+  }
 
   /**
-   * The cost, in fee units, of the reference transaction.
+   * The charge, in drops of XRP, for the reference transaction (This is the transaction cost before scaling for load).
+   *
+   * <p>This field will also be populated with the {@code BaseFee} value from any {@link SetFee} transactions
+   * that occurred before the XRPFees amendment took effect.</p>
+   *
+   * @return An {@link XrpCurrencyAmount}.
+   */
+  @JsonProperty("BaseFeeDrops")
+  @JsonAlias("BaseFee")
+  @JsonDeserialize(using = BaseFeeDropsDeserializer.class)
+  XrpCurrencyAmount baseFeeDrops();
+
+  /**
+   * The cost, in fee units, of the reference transaction. This field can be {@code null} if this {@link SetFee}
+   * transaction was included in a ledger after the XRPFees amendment was enabled because this amendment removes the
+   * {@code ReferenceFeeUnits} field from the {@link SetFee} transaction.
    *
    * @return An {@link UnsignedInteger} cost of ref transaction.
+   *
+   * @deprecated Prefer {@link #maybeReferenceFeeUnits()} over this field.
+   */
+  @Deprecated
+  @Nullable
+  @Value.Default
+  @JsonIgnore
+  default UnsignedInteger referenceFeeUnits() {
+    return maybeReferenceFeeUnits().orElse(null);
```

Comment:
This feels like a breaking change. 

E.g., imagine an application using an older version of xrpl4j  is calling `referenceFeeUnits`. Then, that code upgrades to the version of xrpl4j that has this change (but the application usage of xrpl4j isn't changed because there's just a deprecation warning). If that application starts talking to a server with `XRPFees` amendment enabled, that code will throw a NPE likely.

I need to think more about this one -- unless the argument is that what I wrote above can't happen? Or maybe just is unlikely to happen?

Reply:
I think you're right. Anyone using this method and not checking for null will get a NPE if XRPFees is enabled. Unfortunately I can't think of a way to support the XRPFees change without making this `@Nullable`. We could `@Value.Default` it to `UnsignedInteger.ZERO` to avoid creating NPE, but that almost seems harder to catch as a developer than getting a bunch of NPE in your logs.